### DISCUSSION TITLE
Stand alone ... flash ESP-01S

### DISCUSSION THREAD

**Discussion by Patriboom** [micropython/micropython#9120] [Using MicroPython] [answered]
Hello everyone, 

this topic [https://forum.micropython.org/viewtopic.php?f=16&t=11362](https://forum.micropython.org/viewtopic.php?f=16&t=11362) gave many answers for my project but still ... I need the basic.

How to flash the ESP-01S ?  Have I to do so ? 
How to load my MicroPython ( Thonny / Linux ) into [the tiny board](https://miro.medium.com/max/1104/1*UNAeWfND6nTV_60JFAv-tw.png)

Iv'e bought [this kit](https://www.cafr.ebay.ca/itm/194629833476?_trkparms=amclksrc%3DITM%26aid%3D111001%26algo%3DREC.SEED%26ao%3D1%26asc%3D20160908105057%26meid%3Db41a34f55bb84d0ba0d4d8f17dc2d02f%26pid%3D100675%26rk%3D13%26rkt%3D15%26sd%3D133471810730%26itm%3D194629833476%26pmt%3D1%26noa%3D1%26pg%3D2380057%26brand%3DUnbranded&_trksid=p2380057.c100675.m4236&_trkparms=pageci%3A338c6fb3-226b-11ed-89f3-825ac7fa8724%7Cparentrq%3Ac7b573f31820a45b4a608fddfffda518%7Ciid%3A1) and I want to have it have stand alone in my garage for me to know temperature every morning.  As easy as this.

Maybe, I'll go further and try to transmit images through this tiny wifi antenna.  Who knows  ? 
So, first thing last: 
- have i to flash ESP8266-ESP-01S  to run a thermometer/humidity DHT22 as standalone  ? 
- how to upload code into the module by usign  Thonny under Linux  ? 

Thanks.

**Comment by ThinkTransit** [comment]
I also have a similar module to that but I never put micropython on it.

What you need to do is unplug the ESP part and then connect a usb/serial convertor to the pins.

https://www.ebay.com.au/itm/394206949460?chn=ps&_ul=AU&norover=1&mkevt=1&mkrid=705-139619-5960-0&mkcid=2&itemid=394206949460&targetid=&device=c&mktype=pla&googleloc=9071836&poi=&campaignid=15791083372&mkgroupid=&rlsatarget=&abcId=9300816&merchantid=505771089&gclid=CjwKCAjwgaeYBhBAEiwAvMgp2oa4kQdak4wZIPJGia93jJ8p-ESdYrpUzvT6QSZiQoVEozouJkWneRoCRX0QAvD_BwE

The pins aren't numbered so you will need to figure out which pin is which.

Also make sure you don't put 5v into the module as I think it can only handle 3.3v

There will be a pin that will allow you to put the chip into upload mode that you will need to find.

Once you get that working you can look at flashing micropython.

**Reply by Patriboom** [reply]
As I understand, it's better to use Arduino IDE and to program C++ language than Python or MicroPython.
`What you need to do is unplug the ESP part and then connect a usb/serial convertor to the pins.`
That's the key ... I tought to be able to programm ESP through other card but NO ... I need such kind of convertor.
According to your reply, I found that: [choice of 3 interpretors and how to use it](https://automationcraze.com/esp01-pinout/)

**Comment by ThinkTransit** [comment]
Sorry you shouldn't need to figure out the pins, they are available here.

https://diyprojects.io/esp01-get-started-arduino-platformio-ide-module-choose-pinout/#.YwrrydJBwqI

and you can follow the instructions here to flash micropython

https://docs.micropython.org/en/latest/esp8266/tutorial/intro.html#deploying-the-firmware

**Reply by Patriboom** [reply]
As I did not sit and wait while you were reading my post, I also found this very good and detailed link about 
[how to flash](https://randomnerdtutorials.com/getting-started-thonny-micropython-python-ide-esp32-esp8266/).

**Reply by elschopi** [reply]
First, sorry if this is in the wrong place, I'm still figuring out this "git discussions" thing 😃 
What I found very useful when dealing with ESP-01(S) modules was a USB programmer designed for those. No fear of getting a connection wrong, and can even be used as a short-term supply.
Something like [https://www.amazon.de/AZDelivery-ESP8266-ESP-01-USB-Adapter-Parent/dp/B07ZR9DV3V](url)
However if you use one of these, I'd recommend modifying it with a simple toggle switch for the programming enable GPIO0 as well as a push button for RST. This way, you can program the ESP-01 and then run it.
The only downside is that you need the CH340 usb COM port driver on your system.

**Comment by Patriboom** [comment, marked as answer]
Thanks guys, 

Thanks all, that is very helpfull.
I don't know how to close the topic.  Your replies guide me good.
I wanted to mark as answer both of your text but only one is permitted.  So I'll mark this one (mine) as answer, inviting the coming readers to check up the two previous contributions to find the whole information they need.

**Reply by ThinkTransit** [reply]
You don't need to close it because this is a discussion and not an issue.

Well personally I think micropython is a great starting point as the learning curve isn't as steep. Thonny isn't a bad editor to get started on.

**Reply by Patriboom** [reply]
Thanks.
I'm waiting for my USB - TTL  module.  Then, I'll try those steps and give news.

**Comment by Jibun-no-Kage** [comment]
When I have done this, I have had to make sure for the specific memory chip/type.  I recently had some ESP01s that explicitly needed DOUT mode set, to flash and then boot correctly.  Without setting the mode to DOUT, the flash worked without error, verified, all seemed good.  But when I booted the device.... (what for drum roll..)  Nothing but crickets.  The clue was that when I used the Arduino IDE, I could write and upload a sketch, but until I refreshed the bootloader (via the IDE) after a MP flash attempt.  So I knew something was not being staged right.  So I went back to flashing via Thonny (or esptool) and tried different modes, and found the one that worked.  I can't seem to find much on why this is the case in the official ESP documentation, it seems to be board/vendor specific.

**Reply by Jibun-no-Kage** [reply]
You using ESP01 with 4MB or 1MB?  I am working on a custom firmware that is really lean for 1MB ESP01 modules.  But it is not going to be easy, due to the nature of how MP works on 1MB memory size.  The 1MB size is fine for things like Tasmota and such, but for MP, MP really is easier if you have 4MB modules.
